src/items.rs

```diff
@@ -25,6 +27,54 @@ use syntax::print::pprust;
 use syntax::parse::token;
 
 impl<'a> FmtVisitor<'a> {
+    pub fn visit_let(&mut self, local: &ast::Local, span: Span) {
+        self.format_missing_with_indent(span.lo);
+
+        // New scope so we drop the borrow of self (context) in time to mutably
+        // borrow self to mutate its buffer.
+        let result = {
+            let context = self.get_context();
+            let mut result = "let ".to_owned();
+
+            // 1 = ;
+            // FIXME(#82): reserving space for the semicolon overly conservative
+            // for multi-line expressions.
```

Comment:
First of all, it's great to have you back @cassiersg! I am a bit surprised to find your comment at this particular line. There's no interpretation issue here, is there? I could imagine it at https://github.com/marcusklaas/rustfmt/blob/assignment/src/expr.rs#L143. I believe I made an interpretation error there.

I think of `width` being equal to `MAX_WIDTH - offset - expression_suffix_len`. This would be equal to the length of the last line. The maximum length of the first line would be `MAX_WIDTH - offset`.

This may not be the optimal place to discuss this, but I think we can rewrite the `Rewrite` trait without adding parameters. Instead of passing `width` and `offset`, we may be able to get away with passing `suffix_len` and `offset`. Since all `Rewrite` implementers adhere to the `MAX_WIDTH` limit, we can then compute the widths for the all the lines from these three values.

Also, should we be more economical with our integer sizes in `Rewrite`? `u32` and possibly even `u16` ought to be large enough?